Assets/Scripts/StandardSamples/UI/Friends/UIFriendInteractionSource.cs

```diff
@@ -52,13 +59,29 @@ public virtual void OnFriendInteractButtonClicked(FriendData friendData)
         }
 
         //Should the friend UI update interaction state from this source?
-        public virtual bool IsDirty()
+        public virtual bool IsFriendsUIDirty()
+        {
+            return dirtyFlag;
+        }
+
+        public virtual void ResetFriendsUIDirtyFlag()
         {
-            return false;
+            dirtyFlag = false;
         }
 
-        public virtual void ResetDirtyFlag()
+        public virtual void MarkFriendsUIDirty()
```

Comment:
I believe that this introduces confusion by using the verb "Mark," vs in other instances where this value is referred to a "Flag," which in these contexts are synonymous, but using them interchangeably is (I believe) an inconsistency that can be easily avoided and lend greater clarity to what is happening.

Reply:
What do you think this could be called instead? "Set"FriendsUIDirty?

Reply:
I've always been a fan of doing this:
```csharp
public virtual void SetDirtyFlag(bool isDirty = true)
```

Leaving the `isDirty` parameter to default maintains the correct grammar of "SetDirtyFlag", while also making clear that `SetDirtyFlag(false)` means to _unset_ the dirty flag.

Reply:
That sounds simple enough, I'll do that!

Reply:
There's a message on Slack in relation to this. The subscribed event method is added in this changelist, would like your feedback on naming it.